django/db/models/query.py

```diff
@@ -145,8 +146,7 @@ class FlatValuesListIterable(BaseIterable):
     def __iter__(self):
         queryset = self.queryset
         compiler = queryset.query.get_compiler(queryset.db)
-        for row in compiler.results_iter():
-            yield row[0]
+        return chain.from_iterable(compiler.results_iter())
```

Comment:
Unfortunately I can't say for sure.
It appears that `__getitem__()` it quite slow here as
```
for row in compiler.results_iter():
    value, = row
    yield value
```
works faster than the original version. Perhaps because the former version needs to `LOAD_CONST` and then Python int needs to be converted to C int.

`chain.from_iterable()` version is like that example, but with more things done on C level.